pkg/generate/app/componentresolvers.go

```diff
@@ -299,7 +300,11 @@ func resolveExact(resolver Resolver, value string) (exact *ComponentMatch, inexa
 			return nil, nil, err
 		}
 	}
-	return match, nil, nil
+	if match.Score == 0.0 {
+		return match, nil, nil
+	} else {
```

Comment:
correct.  Because what if all we found for "jenkins" was "jenkins-ransomware-installer"?  You really want us to blindly run the closest match?  I don't.

Or for a more realistic case, what if their app requires ruby-22 but the environment has ruby-20 installed because someone screwed up, and they just specify "ruby"?

also i'm 99% certain we discussed this and you agreed that partial matches should prompt the user, not proceed.

(and it's not new behavior being introduced in this PR, though this PR is making the behavior more consistent and better implemented)